frontend/src/components/managePermissions/UserDataTable.vue

```diff
@@ -16,351 +16,376 @@ import Button from '@/components/common/Button.vue';
+	},
+	isApplicationSelected: {
+		type: Boolean,
+		required: true,
+		default: false,
+	},
 });
 
+const confirm = useConfirm();
+
 const filters = ref({
-    global: { value: null, matchMode: FilterMatchMode.CONTAINS },
-    'user.user_name': { value: null, matchMode: FilterMatchMode.STARTS_WITH },
-    'role.parent_role.role_name': {
-        value: null,
-        matchMode: FilterMatchMode.CONTAINS,
-    },
-    'role.role_name': {
-        value: null,
-        matchMode: FilterMatchMode.CONTAINS,
-    },
-    'role.client_number.forest_client_number': {
-        value: null,
-        matchMode: FilterMatchMode.CONTAINS,
-    },
+	global: { value: null, matchMode: FilterMatchMode.CONTAINS },
+	'user.user_name': { value: null, matchMode: FilterMatchMode.STARTS_WITH },
+	'role.parent_role.role_name': {
+		value: null,
+		matchMode: FilterMatchMode.CONTAINS,
+	},
+	'role.role_name': {
+		value: null,
+		matchMode: FilterMatchMode.CONTAINS,
+	},
+	'role.client_number.forest_client_number': {
+		value: null,
+		matchMode: FilterMatchMode.CONTAINS,
+	},
 });
 
 const confirmDeleteData = reactive({
-    userName: '',
-    role: '',
+	userName: '',
+	role: '',
 });
 
 const emit = defineEmits<emit>();
 
 function deleteAssignment(assignment: FamApplicationUserRoleAssignmentGet) {
-    confirmDeleteData.role = assignment.role.role_name;
-    confirmDeleteData.userName = assignment.user.user_name;
-    confirm.require({
-        group: 'deleteAssignment',
-        header: 'Remove Access',
-        rejectLabel: 'Cancel',
-        acceptLabel: 'Remove',
-        acceptClass: 'p-button-danger',
-        accept: () => {
-            emit('deleteUserRoleAssignment', assignment);
-        },
-    });
+	confirmDeleteData.role = assignment.role.role_name;
+	confirmDeleteData.userName = assignment.user.user_name;
+	confirm.require({
+		group: 'deleteAssignment',
+		header: 'Remove Access',
+		rejectLabel: 'Cancel',
+		acceptLabel: 'Remove',
+		acceptClass: 'p-button-danger',
+		accept: () => {
+			emit('deleteUserRoleAssignment', assignment);
+		},
+	});
 }
 </script>
 
 <template>
-    <ConfirmDialog group="deleteAssignment">
-        <template #message>
-            <p>
-                Are you sure you want to remove
-                <strong>{{ confirmDeleteData.role }}</strong> access to
-                <strong>{{ confirmDeleteData.userName }}</strong> in
-                <strong>{{ selectedApplicationDisplayText }}</strong>
-            </p>
-        </template>
-    </ConfirmDialog>
-    <div class="data-table-container">
-        <div v-if="!props.isApplicationSelected" class="no-app-selected">
-            <IconCapitol />
-            <p class="no-app-selected-title" no-app-selected>
-                Nothing to show yet!
-            </p>
-            <p class="no-app-selected-text">
-                Choose an application to show a list of users with access to it.
-                The list will display here.
-            </p>
-        </div>
```

Comment:
Let's refactor line 99 to line 108 into a new component "TablePlaceholder.vue". And when there is no application selected, in ManagePermission.vue, show the TablePlaceholder. When the user select an application, we then display the tab and the UserDataTable. So you don't have to do style adjustment for tabs based on if application selected or not.

Reply:
I can do that